### DISCUSSION TITLE
Moving JPEG from TIFF container to PDF container

### DISCUSSION THREAD

**Discussion by janist78** [ImageMagick/ImageMagick#7472] [Help] [answered]
### ImageMagick version

7.1.1-Q8

### Operating system, version and so on

Windows 11 x64

### Description

I have JPEG images encapsulated into TIFF (TIFF with JPEG compression) and I need to move it to PDF container (There are multiple pages per TIFF, so I can't use plain JPEG) How to do it to preserve original JPEG source with out re-encoding and loosing quality.
Current command I use is:
magick.exe inputFile.tiff -compress JPEG outputFile.pdf
-compress JPEG is there to fix issue with some TIFF that have different compression
Additionally, if we have to re-encode JPEG, how to set quality equal to input JPEG.

**Comment by snibgo** [comment, marked as answer]
IM will read input images into memory, decompressing them. Then it writes images from memory to disk files, compressing them. It doesn't have a "direct" mechanism that would avoid the decompress and compress processes.

I don't know what software can do what you want. Possibly tiff2pdf.

**Reply by janist78** [reply]
Tested tiff2pdf, in theory it worked, but produced files, when opened, showed blank pages in correct size, despite all data being in file.
So back to original question:
if we have to re-encode JPEG, how to set quality equal to input JPEG.

**Reply by janist78** [reply]
Found this:
magick identify -format '%Q' MyImage
To show compression quality
